core/src/apps/cardano/sign_tx.py

```diff
@@ -533,14 +527,16 @@ async def _process_withdrawals(
     if withdrawals_count == 0:
         return
 
-    # until the CIP with canonical CBOR is finalized storing the seen_withdrawals is the only way we can check for
-    # duplicate withdrawals
-    seen_withdrawals: set[tuple[int, ...] | bytes] = set()
+    previous_credential: tuple[int, ...] | bytes = b""
     for _ in range(withdrawals_count):
         withdrawal: CardanoTxWithdrawal = await ctx.call(
             CardanoTxItemAck(), CardanoTxWithdrawal
         )
-        _validate_withdrawal(withdrawal, seen_withdrawals, signing_mode)
+        previous_credential = _validate_withdrawal(
```

Comment:
Done.

Reply:
Hm, I think I just came up with a better to handle this, sorry. I think we can add a `_derive_withdrawal_reward_address_bytes` (or a function with a similar name) which would return the derived reward address. This function can then be called in `_validate_withdrawal` and also in `_process_withdrawals`. That way the validation (also `cbor.are_canonically_ordered`) can be kept inside of `_validate_withdrawal` and reward address derivation isn't duplicated. I am 80% convinced it's better to place `_derive_withdrawal_reward_address_bytes` in 
 `sign_tx.py` rather than `address.py` but if you strongly otherwise feel free to add it to `address.py`.

Reply:
Ok, I've implemented this new proposal in be57e5c. To be honest, I think it brings some other not so beautiful things (eg. we need to pass more arguments to `_validate_withdrawal`), but whatever.

Reply:
I think it's worth it for having the validation in one place.